src/controllers/apiAuth.controller.ts

```diff
@@ -0,0 +1,62 @@
+import passport from "passport";
+import passportCustom from "passport-custom";
+import bcrypt from "bcrypt";
+const CustomStrategy = passportCustom.Strategy;
+import { pool } from "../config/database";
+
+async function hashPassword(password: string) {
+  const saltRounds = 10;
+  return await bcrypt.hash(password, saltRounds);
+}
+
+passport.use(
+  "custom-api-key",
+  new CustomStrategy(async (req, done) => {
+    try {
+      const phone_number = req.body.phone_number;
+      const password = req.body.password;
+
+      // Check if the user exists
+      const query = {
+        text: "SELECT * FROM celebration.user WHERE phone_number = $1",
```

Comment:
How did it feel writing raw SQL like this?

Reply:
Is it wrong?

Reply:
No, it's not. I just wanted to know how thrilled you felt? 😄